### PR TITLE
Fix DEREF_AFTER_NULL in /src/coreclr/jit/fgopt.cpp

### LINKED ISSUE #99476: DEREF_AFTER_NULL in /src/coreclr/jit/fgopt.cpp
The static analyzer has detected DEREF_AFTER_NULL: after having been compared to a NULL value at fgopt.cpp:1343, pointer '_tryEntryPrev_' is dereferenced at fgopt.cpp:1356.

https://github.com/dotnet/runtime/blob/dc45e96840243b203b13e61952230e225d2aac52/src/coreclr/jit/fgopt.cpp#L1342-L1356

There is a comment from the developers at 1335-1337: it follows that in this code block developers "_rely on the validity of the operation to obtain bbprev (and bbnext)_".

https://github.com/dotnet/runtime/blob/dc45e96840243b203b13e61952230e225d2aac52/src/coreclr/jit/fgopt.cpp#L1335-L1337

Besides, there ia another comment that also expresses hope for the implementation of _'tryEntryPrev != nullptr'_ above the debug assert (at fgopt.cpp:1351).

https://github.com/dotnet/runtime/blob/dc45e96840243b203b13e61952230e225d2aac52/src/coreclr/jit/fgopt.cpp#L1348-L1351

_Dereference of nullptr_ is a serious error that needs to be monitored. 

So if you are sure that _dereference of nullptr_ is impossible here? Why do do check it at fgopt.cpp:1343 **?**

https://github.com/dotnet/runtime/blob/dc45e96840243b203b13e61952230e225d2aac52/src/coreclr/jit/fgopt.cpp#L1343

---

Found by Linux Verification Center (portal.linuxtesting.ru) with SVACE.

Author E. Bykhanova (e.bykhanova@fobos-nt.ru).

### PR DESCRIPTION
**Pull request by EgorBo**
Fixes https://github.com/dotnet/runtime/issues/99476 warning

### CODE REVIEW

**Comment by EgorBo** [comment]
@dotnet/jit-contrib PTAL, trivial fix for the warning. `tryEntryPrev` is never null because we're always expected to have a scratch bb as the top-most bb.

File: src/coreclr/jit/fgopt.cpp [hunk 1]
```diff
@@ -537,7 +537,8 @@ PhaseStatus Compiler::fgPostImportationCleanup()
                 //
                 BasicBlock* const oldTryEntry  = HBtab->ebdTryBeg;
                 BasicBlock*       tryEntryPrev = oldTryEntry->Prev();
-                while ((tryEntryPrev != nullptr) && tryEntryPrev->HasFlag(BBF_REMOVED))
+                assert(tryEntryPrev != nullptr);
+                while (tryEntryPrev->HasFlag(BBF_REMOVED))
                 {
                     tryEntryPrev = tryEntryPrev->Prev();
```
Comment (jakobbotsch) [inline src/coreclr/jit/fgopt.cpp:543]: Add another assert in here at the end?

Comment (jakobbotsch) [reply src/coreclr/jit/fgopt.cpp:543]: (or rather, remove the assert below into the loop)

Comment (EgorBo) [reply src/coreclr/jit/fgopt.cpp:543]: there is already a trailing assert:

image

Comment (EgorBo) [reply src/coreclr/jit/fgopt.cpp:543]: moved it into the loop

**Review by jakobbotsch — APPROVED** [review]

**Pull request merged**